hs_core/views/__init__.py

```diff
@@ -96,6 +99,77 @@ def verify(request, *args, **kwargs):
     return HttpResponseRedirect('/')
 
 
+def get_tasks_by_user(request):
+    if request.user.is_authenticated():
+        user_id = request.user.username
+    else:
+        user_id = request.session.session_key
```

Comment:
I see this if authenticated block several times.  It should be put in a method and called as needed.

Reply:
Good catch and will do.